convert2rhel/cert.py

```diff
@@ -20,56 +20,99 @@
 import os
 import shutil
 
-from convert2rhel import utils
+from convert2rhel import backup, utils
 
 
 loggerinst = logging.getLogger(__name__)
 
 
-class SystemCert(object):
-    def __init__(self):
-        self._target_cert_dir = "/etc/pki/product-default/"
-        self._cert_filename, self._source_cert_dir = self._get_cert()
+class PEMCert(backup.RestorableChange):
+    """Handling certificates needed for verifying Red Hat services."""
+
+    def __init__(self, source_cert_dir, target_cert_dir):
+        super(PEMCert, self).__init__()
+
+        self._target_cert_dir = target_cert_dir
+        self._source_cert_dir = source_cert_dir
+        self._cert_filename = _get_cert(source_cert_dir)
         self._source_cert_path = self._get_source_cert_path()
         self._target_cert_path = self._get_target_cert_path()
 
-    @staticmethod
-    def _get_cert():
-        """Return name of certificate and this directory."""
-        cert_dir = os.path.join(utils.DATA_DIR, "rhel-certs")
-        if not os.access(cert_dir, os.R_OK | os.X_OK):
-            loggerinst.critical("Error: Could not access %s." % cert_dir)
-        pem_filename = None
-        for filename in os.listdir(cert_dir):
-            if filename.endswith(".pem"):
-                pem_filename = filename
-                break
-        if not pem_filename:
-            loggerinst.critical("Error: System certificate (.pem) not found in %s." % cert_dir)
-        return pem_filename, cert_dir
+        self.previously_installed = False
 
     def _get_source_cert_path(self):
         return os.path.join(self._source_cert_dir, self._cert_filename)
 
     def _get_target_cert_path(self):
         return os.path.join(self._target_cert_dir, self._cert_filename)
 
-    def install(self):
+    def enable(self):
         """RHEL certificate (.pem) is used by subscription-manager to
         determine the running system type/version.
         """
```

Comment:
The class is generic - i.e. it's not just for an RHSM product certificate this docstring talks about but also for the cdn.redhat.com SSL CA cert.
```suggestion
        """Install the .pem certificate."""
```